docker-multinode/common.sh

```diff
@@ -0,0 +1,437 @@
+  local BOOTSTRAP_SECONDS=0
+  while [[ $(docker -H ${BOOTSTRAP_DOCKER_SOCK} ps 2>&1 1>/dev/null; echo $?) != 0 ]]; do
+    ((BOOTSTRAP_SECONDS++))
+    if [[ ${BOOTSTRAP_SECONDS} == ${TIMEOUT_FOR_SERVICES} ]]; then
+      kube::log::error "docker bootstrap failed to start. Exiting..."
+      exit
+    fi
+    sleep 1
+  done
+}
+
+# Start etcd on the master node
+kube::multinode::start_etcd() {
+
+  kube::log::status "Launching etcd..."
+  
+  docker -H ${BOOTSTRAP_DOCKER_SOCK} run -d \
+    --restart=${RESTART_POLICY} \
+    --net=host \
+    gcr.io/google_containers/etcd-${ARCH}:${ETCD_VERSION} \
+    /usr/local/bin/etcd \
+      --listen-client-urls=http://127.0.0.1:4001,http://${MASTER_IP}:4001 \
+      --advertise-client-urls=http://${MASTER_IP}:4001 \
+      --data-dir=/var/etcd/data
+
+  # Wait for etcd to come up
+  kube::util::wait_for_url "http://localhost:4001/v2/machines" "etcd" 0.25 80
+
+  # Set flannel net config
+  docker -H ${BOOTSTRAP_DOCKER_SOCK} run \
+      --net=host \
+      gcr.io/google_containers/etcd-${ARCH}:${ETCD_VERSION} \
+      etcdctl \
+      set /coreos.com/network/config \
+          "{ \"Network\": \"${FLANNEL_NETWORK}\", \"Backend\": {\"Type\": \"${FLANNEL_BACKEND}\"}}"
+
+  sleep 2
+}
+
+# Start flannel in docker bootstrap, both for master and worker
+kube::multinode::start_flannel() {
+
+  kube::log::status "Launching flannel..."
+
+  docker -H ${BOOTSTRAP_DOCKER_SOCK} run \
+    --restart=${RESTART_POLICY} \
+    -d \
+    --net=host \
+    --privileged \
+    -v /dev/net:/dev/net \
+    -v ${FLANNEL_SUBNET_TMPDIR}:/run/flannel \
+    quay.io/coreos/flannel:${FLANNEL_VERSION} \
+    /opt/bin/flanneld \
+      --etcd-endpoints=http://${MASTER_IP}:4001 \
+      --ip-masq="${FLANNEL_IPMASQ}" \
+      --iface="${FLANNEL_IFACE}"
+
+  # Wait for the flannel subnet.env file to be created instead of a timeout. This is faster and more reliable
+  local FLANNEL_SECONDS=0
+  while [[ ! -f ${FLANNEL_SUBNET_TMPDIR}/subnet.env ]]; do
+    ((FLANNEL_SECONDS++))
+    if [[ ${FLANNEL_SECONDS} == 20 ]]; then
+      kube::log::error "flannel failed to start. Exiting..."
+      exit
+    fi
+    sleep 1
+  done
+
+  source ${FLANNEL_SUBNET_TMPDIR}/subnet.env
+
+  kube::log::status "FLANNEL_SUBNET is set to: ${FLANNEL_SUBNET}"
+  kube::log::status "FLANNEL_MTU is set to: ${FLANNEL_MTU}"
+}
+
+# Configure docker net settings, then restart it
+kube::multinode::restart_docker(){
+
+  case "${lsb_dist}" in
+    amzn)
+      DOCKER_CONF="/etc/sysconfig/docker"
+
+      kube::helpers::file_replace_line ${DOCKER_CONF} \ # Replace content in this file
+        "--bip" \ # Find a line with this content...
+        "OPTIONS=\"\$OPTIONS --mtu=${FLANNEL_MTU} --bip=${FLANNEL_SUBNET}\"" # ...and replace the found line with this line
+
+      ifconfig docker0 down
+      yum -y -q install bridge-utils 
+      brctl delbr docker0 
+      service docker restart
+      ;;
+    centos)
+      DOCKER_CONF="/etc/sysconfig/docker"
```

Comment:
So `centos 7` uses `systemd`?
Then it's just as easy as do `centos|systemd` on the next case